src/Microsoft.ML.HalLearners/VectorWhitening.cs

```diff
@@ -772,15 +671,143 @@ private static float DotProduct(float[] a, int aOffset, ReadOnlySpan<float> b, R
     /// <include file='doc.xml' path='doc/members/member[@name="Whitening"]/*'/>
     public sealed class VectorWhiteningEstimator : IEstimator<VectorWhiteningTransformer>
     {
+        [BestFriend]
+        internal static class Defaults
+        {
+            public const WhiteningKind Kind = WhiteningKind.Zca;
+            public const float Eps = 1e-5f;
+            public const int MaxRows = 100 * 1000;
+            public const bool SaveInverse = false;
+            public const int PcaNum = 0;
+        }
+
+        /// <summary>
+        /// Describes how the transformer handles one column pair.
+        /// </summary>
+        public sealed class ColumnInfo
+        {
+            /// <summary>
+            /// Name of the column resulting from the transformation of <see cref="InputColumnName"/>.
+            /// </summary>
+            public readonly string Name;
+            /// <summary>
+            /// Name of column to transform. If set to <see langword="null"/>, the value of the <see cref="Name"/> will be used as source.
```

Comment:
>If set to <see langword="null"/>, the value of the <see cref="Name"/> will be used as source. [](start = 45, length = 93)

This part of the comment is only valid for constructor parameter. Here, it does not make sense because `InputColumnName` will always be set in constructor.

Reply:
Thank you for pointing this out Zeeshan!

---
In reply to: [254046435](https://github.com/dotnet/machinelearning/pull/2366#discussion_r254046435) [](ancestors = 254046435)